compiler-core/src/language_server/tests/mod.rs

```diff
@@ -254,6 +269,44 @@ impl ProgressReporter for LanguageServerTestIO {
     }
 }
 
+fn add_hex_path_dep<B>(engine: &mut LanguageServerEngine<LanguageServerTestIO, B>, name: &str) {
+    let compiler = &mut engine.compiler.project_compiler;
+    _ = compiler.config.dependencies.insert(
+        name.into(),
+        Requirement::Hex {
+            version: hexpm::version::Range::new("1.0".to_string()),
+        },
+    );
+    _ = compiler.packages.insert(
+        name.into(),
+        ManifestPackage {
+            name: name.into(),
+            version: Version::new(1, 0, 0),
+            build_tools: vec!["gleam".into()],
+            otp_app: None,
+            requirements: vec![],
+            source: ManifestPackageSource::Hex {
+                outer_checksum: crate::manifest::Base16Checksum(vec![]),
+            },
+        },
+    );
+
+    let toml_path = engine
+        .paths
+        .root()
+        .join("build")
+        .join("packages")
+        .join(name)
+        .join("gleam.toml");
```

Comment:
Please use the method on the path class rather than duplicating this logic here.